lib/sorbet-coerce/converter.rb

```diff
@@ -131,7 +131,7 @@ def _convert_simple(value, type, raise_coercion_error)
     elsif value.is_a?(type)
       return value
     elsif type == BigDecimal
-      return BigDecimal(value)
+      return BigDecimal(value, 0)
```

Comment:
I'm not fully sure this is what we would want 🤔 
```ruby
irb(main):001:0> BigDecimal(123.321, 0).to_digits
=> "123.320999999999997953636921010911464691"
```

maybe we are better off calling `to_s` on the value?
```suggestion
      return BigDecimal(value.to_s)
```
```ruby
irb(main):002:0> BigDecimal(123.321.to_s).to_digits
=> "123.321"
```

Reply:
+1 to `value.to_s`

> we probably still need to check what to do with nils or things that don't respond to #to_s

That's already taken care of on L125.

Reply:
This does not need to be part of this PR, but we should also catch argument errors like
```
ArgumentError (invalid value for BigDecimal(): "a")
```
in this method, so the error can be properly re-raised and being treated properly by the `raise_coercion_error` flag.